server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java

```diff
@@ -66,6 +66,8 @@
     private final NamedWriteableRegistry namedWriteableRegistry;
     private final Function<SearchRequest, InternalAggregation.ReduceContextBuilder> requestToAggReduceContextBuilder;
 
+    private SearchTask searchTask;
```

Comment:
SearchPhaseController is a singleton that is created [here](https://github.com/elastic/elasticsearch/blob/6823b8eb5e02849923e2700bc43f4d392e905f4d/server/src/main/java/org/elasticsearch/node/Node.java#L661), so this instance of the searchTask is going to be shared by all searches in the node leading to a race condition.